segments_config/models.py

```
@@ -0,0 +1,70 @@
+import fnmatch
+
+from io import StringIO
+
+from django.db import models
+from django.core.exceptions import ValidationError
+
+from .validators import alphanum, validate_rules, url_regex
+
+# Create your models here.
+
+
+CHECK_URL_NOT_VALID = "NOT_VALID"
+CHECK_URL_UNKNOWN = "UNKNOWN"
+
+
+def validate_rules_field(value):
+    valid_message = validate_rules(value)
+    if valid_message is not None:
+        raise ValidationError(valid_message)
+
+
+class SegmentsConfig(models.Model):
+
+    name = models.CharField(max_length=50, unique=True, validators=[alphanum])
+    rules_set = models.TextField(verbose_name='Rules set', validators=[validate_rules_field])
+    created = models.DateField(auto_now_add=True, editable=False)
+
+    def __str__(self):
+        return '{} rules set'.format(self.name)
+
+    class Meta:
+        verbose_name = 'segment config'
+        verbose_name_plural = 'segments config'
+
+    @property
+    def config_rules_set(self):
+        like_file = StringIO(self.rules_set)
+        content = iter(like_file.readlines())
+        key = value = None
+
+        for item in content:
+            item = item.strip()
+            if not item:
+                continue
+
+            if key is None:
+                key = item
+                continue
+
+            if value is None:
+                value = item
+
+            assert (key and value)
+
+            yield (key, value)
+            key = value = None
+
+    def check_url(self, url):
+        assert url is not None
+
+        try:
+            url_regex(url)
+        except ValidationError:
+            return CHECK_URL_NOT_VALID
+
+        for key, value in self.config_rules_set:
+            if fnmatch.fnmatchcase(url, value):
+                return key[1:]
+        return CHECK_URL_UNKNOWN
```

you should raise exceptions instead of returning constants. If the user add a segment called "UNKNOWN" you'll have a bug :P